### PR TITLE
Chore improve metatest hooks

### PR DESCRIPTION
**Pull request by flbulgarelli**
This PR allows checker to raise expectations with details. That enables to handle additional test information without having to serialize it and then deserialize - this is for example what gobstones runner currently does - creates a JSON string before raising the exception and then it parses again after handling it, which is both cumbersome and inefficient

### CODE REVIEW

File: spec/mulang_expectations_hook_spec.rb [hunk 1]
```diff
@@ -55,7 +55,7 @@ def language
 
       let(:result) { compile_and_run request }
 
-      it { expect(result.length).to eq 6 }
+      it { expect(result.length).to eq 7 }
```
Comment (flbulgarelli) [inline spec/mulang_expectations_hook_spec.rb:58]: This is an unrelated mulang test stuff

**Comment by julian-berbel** [comment]
> Could we get a test showcasing a usage of render_error_output_with_details?

Nevermind! Just saw that's removed in #104. Not much of a point in testing it ¯\\\_(ツ)_/¯
